src/theory/model_manager_distributed.cpp

```diff
@@ -0,0 +1,76 @@
+/*********************                                                        */
+/*! \file model_manager_distributed.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Management of a distributed approach for model generation.
+ **/
+
+#include "theory/model_manager_distributed.h"
+
+#include "theory/theory_engine.h"
+
+namespace CVC4 {
+namespace theory {
+
+ModelManagerDistributed::ModelManagerDistributed(
+    TheoryEngine& te, EqEngineManagerDistributed& eem)
+    : ModelManager(te), d_eem(eem)
+{
+}
+
+ModelManagerDistributed::~ModelManagerDistributed() {}
+
+bool ModelManagerDistributed::prepareModel()
+{
+  Trace("model-builder") << "ModelManagerDistributed: reset model..."
+                         << std::endl;
+
+  // push/pop to clear the equality engine of the model
+  context::Context* meec = d_eem.getModelEqualityEngineContext();
+  meec->pop();
+  meec->push();
+
+  // Collect model info from the theories
+  Trace("model-builder") << "ModelManagerDistributed: Collect model info..."
+                         << std::endl;
+  // Consult each active theory to get all relevant information concerning the
+  // model, which includes both dump their equality information and assigning
+  // values.
+  for (TheoryId theoryId = theory::THEORY_FIRST; theoryId < theory::THEORY_LAST;
+       ++theoryId)
```

Comment:
What about the order of the theories, e.g. arith before strings?

Reply:
This behavior is preserved in this commit.  This relies on the same thing we've relied on before (regarding the list of theories). I'll add a comment here to mention this.